packages/web-api-scan-runner/src/crawl-runner/crawl-runner.ts

```diff
@@ -19,14 +18,12 @@ export class CrawlRunner {
         @inject(crawlerIocTypes.CrawlerProvider) private readonly getCrawler: CrawlerProvider,
         @inject(GlobalLogger) private readonly logger: GlobalLogger,
         @inject(ServiceConfiguration) private readonly serviceConfig: ServiceConfiguration,
-        @inject(ScanMetadataConfig) scanMetadataConfig: ScanMetadataConfig,
         @inject(BatchConfig) private readonly batchConfig: BatchConfig,
-    ) {
-        const scanMetadata = scanMetadataConfig.getConfig();
-        this.logger.setCommonProperties({ scanId: scanMetadata.id });
```

Comment:
You've removed this line, but didn't replace it. There's still no guarantee this will be set before it's required.